### PR TITLE
Users can view a particular Gif, like and comment on any article or gifs

### PR DESCRIPTION
**Pull request by goodnessaig1**

### CODE REVIEW

File: src/Auth/Actions/articleActions.js [hunk 1]
```diff
@@ -81,3 +139,73 @@ export function PostArticles(credentials) {
...
+            function (error) {
+                const errorMsg = error;
+                dispatch(likeArticlesFailure(errorMsg));
+            }
+        );
+        return promise;
+    };
+}
+
+export function GetSingleArticle(id, setArticleModal) {
```
Comment (courage173) [inline src/Auth/Actions/articleActions.js]: why will you pass this here. please do it inside the component

File: src/Auth/Actions/articleActions.js [hunk 2]
```diff
@@ -81,3 +139,73 @@ export function PostArticles(credentials) {
...
+                }
+                setArticleModal(false);
+                dispatch(getSingleArticleFailure(errorMsg));
+            }
+        );
+        return promise;
+    };
+}
+
+export function PostArticleComment(data, Id) {
```
Comment (courage173) [inline src/Auth/Actions/articleActions.js]: why is Id capital letter?

File: src/Auth/Actions/gifActions.js [hunk 3]
```diff
@@ -46,3 +105,71 @@ export const PostGif = (formData) => {
...
+                    });
+                }
+                setGifModal(false);
+                dispatch(getSingleGifFailure(errorMsg));
+            }
+        );
+        return promise;
+    };
+}
+export function PostGifComment(data, gifId, index) {
```
Comment (courage173) [inline src/Auth/Actions/gifActions.js]: why are you passing index? I don't get

File: src/Components/Dashboard/ArticleCommentModal.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    user,
+    setArticleModal,
+    comments,
+    PostArticleComment,
+    postCommentRequest,
+}) => {
+    const articleId = articleData?.postid;
+    const messagesEndRef = useRef(null);
+    useEffect(() => {
+        setTimeout(() => {
```
Comment (courage173) [inline src/Components/Dashboard/ArticleCommentModal.js]: why do you do this??

**Review by courage173 — CHANGES_REQUESTED** [review]

File: src/Auth/Actions/articleActions.js [hunk 5]
```diff
@@ -41,6 +60,64 @@ export const PostArticlesFailure = (error) => {
     };
 };
 
+export const likeArticlesRequest = () => {
+    return {
+        type: types.LIKE_ARTICLES_REQUEST,
+    };
+};
+
+export const likeArticlesSuccess = (index, likedArticle) => {
```
Comment (courage173) [inline src/Auth/Actions/articleActions.js]: what is index doing here, why don't you use the article id

File: src/Auth/Reducer/feedsReducer.js [hunk 6]
```diff
@@ -67,6 +67,42 @@ export default function (state = initialState, action) {
...
+            updatedFeeds[index] = likedArticle;
+            return Object.assign({}, state, {
+                ...state,
+                allFeeds: updatedFeeds,
+            });
+
+        case types.LIKE_GIF_SUCCESS:
+            const { indexNumber, likedGif } = action.payload;
+            updatedFeeds = [...state.allFeeds];
+            updatedFeeds[indexNumber] = likedGif;
```
Comment (courage173) [inline src/Auth/Reducer/feedsReducer.js]: this is very wrong. you can pass the gif id, look for the index using findIndex on the feeds and then you can continue

File: src/Components/Dashboard/Feeds.js [hunk 7]
```diff
@@ -6,25 +6,42 @@ import {
...
     setOffSet,
     feedsLength,
     requesting,
     feedsTotal,
+    clickedImage,
+    setClickedImage,
+    open,
+    setOpen,
 }) => {
+    console.log(feeds);
```
Comment (courage173) [inline src/Components/Dashboard/Feeds.js]: console

**Review by courage173 — CHANGES_REQUESTED** [review]
fix this issues and merge

File: src/Auth/Reducer/feedsReducer.js (hunk 6 above)
Comment (goodnessaig1) [reply src/Auth/Reducer/feedsReducer.js]: Ok sir. i have made the changes using findIndex()

File: src/Auth/Actions/articleActions.js (hunk 5 above)
Comment (goodnessaig1) [reply src/Auth/Actions/articleActions.js]: ok sir

File: src/Components/Dashboard/Feeds.js (hunk 7 above)
Comment (goodnessaig1) [reply src/Components/Dashboard/Feeds.js]: It was a mistake sir, but i have remove it now

**Pull request merged**